IntegrationTests/tests_04_performance/test_01_resources/test_1000_udpconnections.swift

```diff
@@ -14,17 +14,37 @@
 
 import NIO
 
-fileprivate final class DoNothingHandler: ChannelInboundHandler {
+fileprivate final class CountReadsHandler: ChannelInboundHandler {
     public typealias InboundIn = ByteBuffer
     public typealias OutboundOut = ByteBuffer
+    
+    private var readsRemaining: Int
+    private let completed: EventLoopPromise<Void>
+    
+    var completionFuture: EventLoopFuture<Void> { return self.completed.futureResult }
+    
+    init(numberOfReadsExpected: Int, eventLoop: EventLoop) {
```

Comment:
Feels weird passing in the `EventLoop` just to make a promise; why not just pass in a promise?

Reply:
I did ponder it that way round but felt that it's the handler that's "making the promise".  That being said, I don't mind changing.

Reply:
I guess it matches the passing in of promise to the close methods above so I'll change to mirror.